ark/phenotyping/som_runner.R

```diff
@@ -0,0 +1,34 @@
+library(data.table)
+library(FlowSOM)
+library(rhdf5)
+
+print("Reading command args")
+args <- commandArgs(trailingOnly=TRUE)
+
+# get path to pixel mat data
+print("Getting pixel mat data path")
+pixelMatPath <- args[1]
+
+# create a vector out of the list of channels provided
+print("Creating marker list")
+markers <- unlist(strsplit(args[2], split=","))
+
+# get the path to the write directory
+print("Getting the write directory")
+write_dir <- args[3]
+
+# read the pixel mat data
+print("Reading the pixel matrix data")
+pixelMatData <- fread(pixelMatPath, select=markers)
+
+# TODO: subset the data for SOM clustering only on the channels we're interested in
+
+# run the SOM clustering step
+print("Run the SOM clustering")
+som_results <- SOM(data=as.matrix(pixelMatData))
```

Comment:
Now that we know this sanity check works, is there a way to access the hidden function SOM uses purely for training, rather than calling the entire FlowSOM process?

Reply:
So `SOM` only does the training. The other stuff (ex. meta-clustering) are handled by separate calls.

Reply:
Okay got it. I assumed based on your comment below ("add original cluster labels to pixelMatData and save, used for verification"), that both the ```SOM``` call and the ```MapDataToCodes``` function were producing labels for for each row of the matrix

Reply:
OK, I think I understand. `SOM` will automatically assign labels to the dataset passed in because it makes a call to `MapDataToCodes`. We need to make a separate call to `MapDataToCodes` because we don't want this default functionality of training and clustering on the same dataset, but rather only want to use the trained weights returned by `SOM` (which is done on a subset) and run `MapDataToCodes` separately on the entire dataset.

Ideally, we don't want this extra `MapDataToCodes` call during the call to `SOM`; unfortunately, as of the current version, this is default functionality. I can talk with Sophie and Sam at FlowSOM to see if it's possible to do this.

Reply:
No problem, it's not a big deal. If there's not already a way to do it we can put this on the backburner